tauri/src/error.rs

```diff
@@ -46,6 +46,12 @@ pub enum Error {
   /// Invalid args when running a command.
   #[error("invalid args for command `{0}`: {1}")]
   InvalidArgs(&'static str, serde_json::Error),
+  /// Encountered an event string that didn't parse into a correct event
+  #[error("raw event string doesn't parse to an event type: {0}")]
+  InvalidEvent(String),
```

Comment:
this isn't used right now; are we going to use this event kind instead of a panic on event parse?

Reply:
Probably not, we can just panic with an error. If someone has a custom event it will be their own error type that is panicked

Reply:
For the docs about having your own error type, we should make it clear that they will want to have a variant for "unknown" ones because of the events both js and tauri itself can generate.  Even better is if we can expose all our events as an enum that meets the requirements of `Tag` and let them have a `Tauri(TauriEvent)` (or w/e name) variant for easy official tauri typings